### DISCUSSION TITLE
FastAPI CLI does not support the dot/colon path that uvicorn supports

### DISCUSSION THREAD

**Discussion by pamelafox** [fastapi/fastapi#11540] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
We previously used this command for local testing:


uvicorn src.api.main:app --port 3100 --reload


I found I had to change it to a path like:

```shell
fastapi dev src/api/main.py
```

I think it may be worth calling out the difference in your documentation, given many developers will be porting from uvicorn to fastapi command.

I'm also not sure if `fastapi dev` will accept an app (versus a path), as I've started using editable app installs in many projects. I haven't tried it in one of those projects yet.

### Description

See above.

### Operating System

Linux

### Operating System Details

Debian Dev Container in VS Code

### FastAPI Version

0.111.0

### Pydantic Version

2.7.1

### Python Version

Python 3.11.9

### Additional Context

Repo is here https://github.com/pamelafox/simple-fastapi-container for a specific example, but this applies to any repos that were using the dot/colon syntax.

**Comment by YuriiMotov** [comment, marked as answer]
This is now supported with `--entrypoint` option:

```
fastapi run --entrypoint some.importable_module:app_name
```
or

```
fastapi run -e some.importable_module:app_name
```


See:
* https://github.com/fastapi/fastapi-cli/releases/tag/0.0.11
* https://github.com/fastapi/fastapi-cli/releases/tag/0.0.15
